pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
The issue of extensibility across quota, limitrange, UI, etc., is beyond just the APIs. I'd recommend deferring it since it requires changes to the rest of the system as well.